lib/manageiq/appliance_console/database_admin.rb

```diff
@@ -55,13 +54,14 @@ def activate
       end
 
       def ask_file_location
-        case @backup_type = ask_with_menu(*file_menu_args)
-        when LOCAL_FILE then ask_local_file_options
-        when NFS_FILE   then ask_nfs_file_options
-        when SMB_FILE   then ask_smb_file_options
-        when S3_FILE    then ask_s3_file_options
-        when FTP_FILE   then ask_ftp_file_options
-        when CANCEL     then raise MiqSignalError
+        @backup_type = ask_with_menu(*file_menu_args) do |menu|
+          menu.choice(CANCEL) { |_| raise MiqSignalError }
```

Comment:
I wonder if this is something we should just add to the end of every menu? Unlikely that it should be added to this PR, but this made me think of it.

Reply:
@kbrock When I was typing this I was thinking of modifying ask_with_menu to add that as an option

Reply:
Maybe just make an issue for this?  I agree, that would definitely make this cleaner.  Maybe also have it as an option to opt of ( something like `ask_with_menu(*file_menu_args, :cancel_option => false)` maybe?).

...which reminds me... I really need to add an issue for the "UI stuff" I have wanted to refactor in this...